internal/controller/log.go

```diff
@@ -48,15 +51,25 @@ func (d *DeployLog) LogDeploy(timeStamp time.Time, podName string, podIP string,
 		Reason:           reason,
 	}
 
-	d.Entries = append(d.Entries, newEntry)
+	if len(d.entries) >= d.maxEntries {
+		// Pull elements off the front of the slice to make sure that the newly appended record is under the max record value.
+		d.entries = d.entries[(len(d.entries)-d.maxEntries)+1:]
+	}
+
+	d.entries = append(d.entries, newEntry)
 }
 
 // GetLog returns a json representation of the entries list.
 func (d *DeployLog) GetLog() []byte {
```

Comment:
That's more a general remark. But I would remove the serialization logic from here and move it to the API handler to return a  `[]Entry` here.

Only the API needs this data as JSON, and I don't like having API logic in this common object that one could want to reuse at some point.  This reduces reusability.